util/HalideTraceViz.cpp

```diff
@@ -678,78 +677,73 @@ int run(int argc, char **argv) {
                 fi.stats.observe_load(p);
             }
 
-            // Check the tracing packet contained enough information
-            // given the number of dimensions the user claims this
-            // Func has.
-            assert(p.dimensions >= p.type.lanes * fi.config.strides.size());
-            if (p.dimensions >= p.type.lanes * fi.config.strides.size()) {
-                for (int lane = 0; lane < p.type.lanes; lane++) {
+            // zero- or one-dimensional Funcs can have dimensions < strides
```

Comment:
I don't understand how this is possible. Shouldn't a zero-d function have zero strides, and a one-d function have one Point?

Reply:
Strides are provided by the `--stride` flag, so they can be anything. If you don't specify them at all, they default to `{{1,0},{0,1}` (aka size=2). 

(If the strides were provided by the trace instrumentation itself, then your assertion would be correct.)

Reply:
(I'll add this as a comment since clearly it's possibly confusing)